index.bs

```diff
@@ -3038,29 +3032,31 @@ The [=remote end steps=] with |command parameters| are:
   1. If the implementation is unable to create a new browsing context for any
      reason then return [=error=] with [=error code=] [=unsupported operation=].
 
-  1. Let |user context| be the [=default user context=] if |reference context| is null, and |reference context|' [=user context=] otherwise.
+  1. Let |user context| be the [=default user context=] if |reference context|
+     is null, and |reference context|' [=associated user context=] otherwise.
 
   1. If |command parameters|[<code>userContext</code>] is present and is not null:
 
     1. Let |user context id| be |command parameters|[<code>userContext</code>].
 
-    1. If the [=user context=] with the [=user context id=] equal to |user context id| does
+    1. If the [=user context=] with the [=user context id=] |user context id| does
       not exist in the [=remote end=], return [=error=] with [=error code=] [=no such user context=].
 
-    1. Set |user context| to the [=user context=] with the [=user context id=] equal
-      to |user context id|.
+    1. Set |user context| to the [=user context=] with the [=user context id=] |user context id|.
```

Comment:
I suggest rewriting this as:

```
1. Let |found context| be false.

1. For each |target user context| in the [=set of user contexts=]:

  1. If |target context|'s [=user context id=] equals |command parameters|["<code>userContext</code>"]:

    1. Set |user context| to |target context|.
    
    1. Let |found context| be true.
    
    1. Break.
 
 1. If |found context| is false, return [=error=] with [=error code=] [=no such user context=].
 ```
 
 We could also break this out into a "get user context" algorithm that returns null if there's no context, like we do for browsing contexts.

Reply:
I have extracted the get user context algorithm